src/components/NetworkAssetSelectInput/NetworkAssetSelectInput.tsx

```diff
@@ -318,6 +368,30 @@ const NetworkAssetSelectInput = ({
     setAssetSearchQuery('');
   }
 
+  const formatBalanceByChainByAccountType = (
+    supportedChain: Chain,
+    accType?: string
+  ) => {
+    if (
+      accType === DestinationWalletEnum.Contract &&
+      label === 'From' &&
+      balancePerChainSmartWallet &&
+      balancePerChainSmartWallet.length
+    ) {
+      let balanceByChain = balancePerChainSmartWallet.filter(
+        (item: any) => item.chain === supportedChain.chainId
+      );
+      return balancePerChainSmartWallet &&
+        balancePerChainSmartWallet.length &&
+        balanceByChain.length
+        ? formatAmountDisplay(String(balanceByChain[0].total), '$')
+        : '$0';
+    } else if (accType === DestinationWalletEnum.Key && label === 'From') {
+      return '$0';
+    }
+    return '$0';
```

Comment:
```suggestion
    }
    return '$0';
```

Reply:
So I kept the eleif condition there for keybased and it has been handled there in another PR